bleak/backends/corebluetooth/PeripheralDelegate.py

```diff
@@ -369,6 +380,19 @@ def did_update_value_for_characteristic(
     ) -> None:
         c_handle = characteristic.handle()
 
+        if not error:
```

Comment:
We could probably also bring the `if not future` check up here too to avoid repeating some code.

Reply:
Rearranged!  I also added in a check for the case of 'no future' below this, since pulling that up still left it open for the case of "no future, but an error = call future and fail"